flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPool.java

```diff
@@ -324,78 +297,98 @@ public void disconnectResourceManager() {
 			boolean allowQueuedScheduling,
 			Time allocationTimeout) {
 
-		log.debug("Received slot request [{}] for task: {}", slotRequestId, task.getTaskToExecute());
-
-		final SlotSharingGroupId slotSharingGroupId = task.getSlotSharingGroupId();
-
-		if (slotSharingGroupId != null) {
-			// allocate slot with slot sharing
-			final SlotSharingManager multiTaskSlotManager = slotSharingManagers.computeIfAbsent(
-				slotSharingGroupId,
-				id -> new SlotSharingManager(
-					id,
-					this,
-					providerAndOwner));
-
-			final SlotSharingManager.MultiTaskSlotLocality multiTaskSlotLocality;
-
-			try {
-				if (task.getCoLocationConstraint() != null) {
-					multiTaskSlotLocality = allocateCoLocatedMultiTaskSlot(
-						task.getCoLocationConstraint(),
-						multiTaskSlotManager,
-						slotProfile,
-						allowQueuedScheduling,
-						allocationTimeout);
+		return CompletableFuture.completedFuture(null).thenComposeAsync((i) -> {
```

Comment:
In the end yes, absolutely right. But this is currently on purpose together with some of your other comments that there is still slot pool code around that could be removed. The purpose is you can switch back and forth now between the new and the old code path by changing a single line in the job master and to make the change as incremental as possible, as we previously discussed.

Reply:
Stefan and me had an offline discussion and we concluded to remove now the porting scaffolds because they are no longer needed. The `Scheduler` should now be responsible for the `LogicalSlot` allocation.